src/core/lib/surface/call.c

```diff
@@ -1087,6 +1089,24 @@ static void receiving_initial_metadata_ready(grpc_exec_ctx *exec_ctx,
         &call->metadata_batch[1 /* is_receiving */][0 /* is_trailing */];
     grpc_metadata_batch_filter(md, recv_initial_filter, call);
 
+    /* make sure the received grpc-encoding is amongst the ones listed in
+     * grpc-accept-encoding */
+
+    GPR_ASSERT(call->encodings_accepted_by_peer != 0);
```

Comment:
Is there a possibility that this assert() might break with old clients?  (for whatever reason someone running a client with older version of gRPC but latest server) or is `call->encodings_accepted_by_peer` supposed to be !=0 at this point ?